activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceController.java

```diff
@@ -0,0 +1,73 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.services.query.rest;
+
+import java.util.Optional;
+
+import com.querydsl.core.types.Predicate;
+import org.activiti.cloud.services.query.app.repository.ProcessInstanceRepository;
+import org.activiti.cloud.services.query.model.ProcessInstance;
+import org.activiti.cloud.services.query.resources.ProcessInstanceResource;
+import org.activiti.cloud.services.query.rest.assembler.ProcessInstanceResourceAssembler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.querydsl.binding.QuerydslPredicate;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/v1/" + ProcessInstanceRelProvider.COLLECTION_RESOURCE_REL , produces = MediaTypes.HAL_JSON_VALUE)
+public class ProcessInstanceController {
+
+    private final ProcessInstanceRepository processInstanceRepository;
+
+    private ProcessInstanceResourceAssembler processInstanceResourceAssembler;
+
+    private PagedResourcesAssembler<ProcessInstance> pagedResourcesAssembler;
+
+    @Autowired
+    public ProcessInstanceController(ProcessInstanceRepository processInstanceRepository,
+                                     ProcessInstanceResourceAssembler processInstanceResourceAssembler,
+                                     PagedResourcesAssembler<ProcessInstance> pagedResourcesAssembler) {
+        this.processInstanceRepository = processInstanceRepository;
+        this.processInstanceResourceAssembler = processInstanceResourceAssembler;
+        this.pagedResourcesAssembler = pagedResourcesAssembler;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public PagedResources<ProcessInstanceResource> findAll(@QuerydslPredicate(root = ProcessInstance.class) Predicate predicate,
+                                                           Pageable pageable) {
+        return pagedResourcesAssembler.toResource(processInstanceRepository.findAll(predicate,
+                                                                           pageable),
+                                                  processInstanceResourceAssembler);
+    }
+
+    @RequestMapping(value = "/{processInstanceId}", method = RequestMethod.GET)
+    public ProcessInstanceResource findById(@PathVariable String processInstanceId) {
+        Optional<ProcessInstance> findResult = processInstanceRepository.findById(processInstanceId);
```

Comment:
You could use [EntityFinder](https://github.com/Activiti/activiti-cloud-query-service/blob/develop/activiti-cloud-services-query/activiti-cloud-services-query-repo/src/main/java/org/activiti/cloud/services/query/app/repository/EntityFinder.java) here. Unless you think it's clearer in the way it is now:
```
entityFinder.findById(processInstanceRepository, 
                                    processInstanceId,
                                    "Unable to find processInstance for the given id:'" + processInstanceId + "'");
```
Replaces

```
Optional<ProcessInstance> findResult = processInstanceRepository.findById(processInstanceId);
if (!findResult.isPresent()) {
        throw new RuntimeException("Unable to find processInstance for the given id:'" + processInstanceId + "'");
    }
    return processInstanceResourceAssembler.toResource(findResult.get());
}
```

Reply:
yes I don't see much difference, on a developer perspective i would understand more the repository.find() signature directly if i am reading that code for the first time. If we would need to wrap the repository on something in the future we could create @Service classes.

Reply:
as written above i will use it as the response code of the runtime exception thrown in the controller is not correct.

One more thing, looking at the implementation of EntityFinder there is no reason to be a @Component class as the repository is passed as argument, should we change after this PR?